sites/partners/__tests__/pages/applications/index.test.tsx

```diff
@@ -0,0 +1,203 @@
+import React from "react"
+import { fireEvent } from "@testing-library/react"
+import { rest } from "msw"
+import { setupServer } from "msw/node"
+import { application, user, listing } from "@bloom-housing/shared-helpers/__tests__/testHelpers"
+import {
+  Listing,
+  ListingsStatusEnum,
+} from "@bloom-housing/shared-helpers/src/types/backend-swagger"
+import ApplicationsList from "../../../src/pages/listings/[id]/applications/index"
+import { mockNextRouter, render } from "../../testUtils"
+
+const server = setupServer()
+
+beforeAll(() => {
+  server.listen()
+})
+
+afterEach(() => {
+  server.resetHandlers()
+  window.sessionStorage.clear()
+})
+
+afterAll(() => {
+  server.close()
+})
+
+describe("applications", () => {
+  it("should render error text when the api call fails", async () => {
+    mockNextRouter({ id: "Uvbk5qurpB2WI9V6WnNdH" })
+
+    server.use(
+      rest.get("http://localhost:3100/applications/list", (_req, res, ctx) => {
+        return res(ctx.status(500), ctx.json(""))
+      })
+    )
+    const { findByText } = render(<ApplicationsList />)
+
+    const error = await findByText("An error has occurred.")
+    expect(error).toBeInTheDocument()
+  })
+
+  it("should render applications table when data is returned", async () => {
+    mockNextRouter({ id: "Uvbk5qurpB2WI9V6WnNdH" })
+
+    server.use(
+      rest.get("http://localhost/api/adapter/listings/Uvbk5qurpB2WI9V6WnNdH", (_req, res, ctx) => {
+        return res(ctx.json(listing))
+      }),
+      rest.get("http://localhost/api/adapter/applications", (_req, res, ctx) => {
+        return res(ctx.json({ items: [application], meta: { totalItems: 1, totalPages: 1 } }))
+      }),
+      rest.get("http://localhost/api/adapter/applicationFlaggedSets", (_req, res, ctx) => {
+        return res(ctx.json({ items: [], meta: { totalItems: 0, totalPages: 0 } }))
+      }),
+      rest.get("http://localhost/api/adapter/applicationFlaggedSets/meta", (_req, res, ctx) => {
+        return res(ctx.json({ totalCount: 1 }))
+      })
+    )
+    const { findByText, getByText, getAllByText } = render(<ApplicationsList />)
+
+    const header = await findByText("Partners Portal")
+    expect(header).toBeInTheDocument()
+
+    expect(getAllByText("Archer Studios").length).toBeGreaterThan(0)
+    expect(getByText("Add Application")).toBeInTheDocument()
+    expect(getByText("Export")).toBeInTheDocument()
+    expect(getAllByText("All Applications").length).toBeGreaterThan(0)
+
+    expect(getByText("Application Type")).toBeInTheDocument()
```

Comment:
Love this test. Historically tables have been fairly hard to accurately test that all of the rows have the data shown in the right order (the data is in the column it is supposed to be), especially when using a third party table library. There are ways to do it such as something like this:
```
    expect(
      getByRole("row", {
        name: "6/24/2024 at 5:49:46 PM UTC Electronic Applicant First Applicant Last 2 $40,000 Yes None 10/10/1990 first.last@example.com (123) 123-1231 Home (456) 456-4564 Cell 3200 Old Faithful Inn Rd Yellowstone National Park WY 82190 1000 US-36 Estes Park C0 80517 9035 Village Dr Yosemite Valley CA 95389 Alternate First Alternate Last Case manager or housing counselor Alternate Agency alternate@email.com (789) 012-3456 25 Visitor Center Rd Bay Harbor ME 04609 Household First Household Last 11/25/1966 Friend Yes Yes n/a No",
      })
    ).toBeInTheDocument()
```
However, that is also not perfect for our use case since the date and times are dynamic (that example would fail because it won't be 5:49:46 PM next time it is run).
So this is just a long way of saying that what you have here is good enough. And I'm more excited to have the test at all!